### PR TITLE
Stuff for CI

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Forward-ports of the last batch, another random fix I stumbled across, and some small new features. I'll probably merge once CI passes.

### CODE REVIEW

File: src/scripting/game_lua_kernel.cpp [hunk 1]
```diff
@@ -1424,6 +1424,134 @@ static int impl_end_level_data_collect(lua_State* L)
...
+	} else {
+		char const *m = luaL_checkstring(L, 2);
+		return_string_attrib("scenario", settings.name); // TODO: Deprecate this -> mp_settings.game_name
+		return_string_attrib("game_name", settings.name);
+		return_string_attrib("hash", settings.hash);
+		return_string_attrib("mp_era_name", settings.mp_era_name); // TODO: Deprecate this -> scenario.era.name
+		return_string_attrib("mp_scenario", settings.mp_scenario); // TODO: Deprecate this -> scenario.id
+		return_string_attrib("mp_scenario_name", settings.mp_scenario_name); // TODO: Deprecate this -> scenario.name (which doesn't exist yet)
+		return_string_attrib("mp_campaign", settings.mp_campaign); // TODO: Deprecate this -> scenario.campaign.id
+		return_string_attrib("side_users", utils::join_map(settings.side_users)); // TODO: Deprecate this -> mp_settings.side_players
```
Comment (stevecotton) [inline src/scripting/game_lua_kernel.cpp:1468]: Most of these todos look like more "moving stuff because you don't like the current name".

**Comment by stevecotton** [comment]
> Forward-ports of the last batch, another random fix I stumbled across, and some small new features.

Why are all of those in a single PR?

**Comment by CelticMinstrel** [comment]
It's not a pull request. It's a CI run.

File: src/scripting/game_lua_kernel.cpp (hunk 1 above)
Comment (Pentarctagon) [reply src/scripting/game_lua_kernel.cpp:1468]: I don't want these deprecated.  They're supposed to still exist and people might be using them, so just leave them be.

**Pull request merged**